starknet_py/abi/v1/parser_transformer.py

```diff
@@ -0,0 +1,65 @@
+import os
+
+import lark
+from lark import Transformer, v_args
+
+from starknet_py.cairo.data_types import ArrayType, CairoType, FeltType, Option
+
+
+class ParserTransformer(Transformer):
+    """
+    Transforms the lark tree into a CairoTypes.
+    """
+
+    # pylint: disable=no-self-use
+
+    def __default__(self, data: str, children, meta):
+        raise TypeError(f"Unable to parse tree node of type {data}")
+
+    @v_args(inline=True)
+    def start(self, value):
+        return value
+
+    def type(self, value):
+        return value[0]
+
+    def type_felt(self, _value):
+        return FeltType()
+
+    def type_uint(self, _value):
+        return FeltType()
```

Comment:
I think treating uint as felt is not a good approach. Larger uints will most likely need to be represented as multiple felts.